jupysql_plugin/widgets/connector_widget.py

```diff
@@ -132,28 +78,29 @@ def _handle_message(self, widget, content, buffers):
             elif method == "submit_new_connection":
                 new_connection_data = content["data"]
                 connection_name = new_connection_data.get("connectionName")
-                is_unique_name = _is_unique_connection_name(connection_name)
 
-                if not is_unique_name:
+                try:
+                    connection_name = (
+                        self.widget_manager.save_connection_to_config_file_and_connect(
+                            new_connection_data
+                        )
+                    )
+                except exceptions.ConnectionWithNameAlreadyExists:
                     self.send(
```

Comment:
Currently, `send_error_message_to_front` sends the same method to the client so we can't re-use it. we don't have a lot of errors now but this might be difficult to manage when we add more custom exceptions (using `self.send` might be confusing)

We can modify `send_error_message_to_front` so it will receive also `method` and we'll manage all client errors in the same method. I think we can add a todo here so next time we add another exception we should modify this method as well.

Reply:
I'm not following, if `exceptions.ConnectionWithNameAlreadyExists` happens, then we call self.send with method = "connection_name_exists_error", on the other hand, `send_error_message_to_front` calls with method = "connection_error", I don't see why you're mentioning that they send the same method

Reply:
I'm sorry, I wasn't explaining myself correctly. I meant that the method in `send_error_message_to_front` is hardcoded so the function always sends `connection_error` and we can modify it to something like this: `def send_error_message_to_front(self, method, error)`

This way the function is responsible for managing all errors we want to show to the client (i.e. formatting the error message), and sending them to the kernel using `self.send`.

Reply:
got it. done https://github.com/ploomber/jupysql-plugin/pull/70/commits/7960c1381e35aa3cbe6d764aaa470dac6bcdb565